### PR TITLE
Renamed chainId with chain wherever necessary

### LINKED ISSUE #68: Rename chainId to chain.
This is a followup ticket of #66

The `./mosaic` commands accept the chainId in input, but the value for this can be chainId as well as official chain name. 
In the code, its referred to chainId and its a bit confusing as actual chain id has its own significance and is used to assign the ports.

So instead of referring this to `chainId` we can rename it to the more generic name `chain`. 

Dont update all the instances of `chainId`, but identify if the value can be chainId or chain name, If its just chainId then dont change it.

### PR DESCRIPTION
**Pull request by sarvesh-ost**
fixes #68  

This PR renames `chainId` -> `chain` where ever necessary.

### CODE REVIEW

File: src/bin/mosaic-stop.ts [hunk 1]
```diff
@@ -9,9 +9,9 @@ import Node from '../Node/Node';
 mosaic
   .version(version)
   .arguments('<chains...>')
-  .action((chainIds: string[]) => {
-    for (const chainId of chainIds) {
-      const chain: Node = NodeFactory.create(new NodeDescription(chainId));
+  .action((chains: string[]) => {
+    for (const currentChain of chains) {
```
Comment (schemar) [inline src/bin/mosaic-stop.ts]:
```suggestion
    for (const chain of chains) {
```

File: src/bin/mosaic-stop.ts [hunk 2]
```diff
@@ -9,9 +9,9 @@ import Node from '../Node/Node';
...
+      const chain: Node = NodeFactory.create(new NodeDescription(currentChain));
```
Comment (schemar) [inline src/bin/mosaic-stop.ts]:
```suggestion
      const node: Node = NodeFactory.create(new NodeDescription(chain));
```

File: src/bin/mosaic-stop.ts [hunk 3]
```diff
@@ -9,9 +9,9 @@ import Node from '../Node/Node';
...
 
       chain.stop();
```
Comment (schemar) [inline src/bin/mosaic-stop.ts]:
```suggestion
       node.stop();
```

File: src/Node/GethNode.ts [hunk 4]
```diff
@@ -85,7 +85,7 @@ export default class GethNode extends Node {
 
     args = args.concat([
       'ethereum/client-go:v1.8.23',
-      '--networkid', this.chainId,
+      '--networkid', this.chain,
```
[the remark this answers is not in the document]
Comment (sarvesh-ost) [reply src/Node/GethNode.ts]: Yea, I agree readme should specify details about this.

**Comment by sarvesh-ost** [comment]
> Mostly, the changes look good to me 🙌 But two places I think there should also be changes (for consistency):
> 
> * `mosaic-create`

If I understand it correctly. `mosaic-create` command expects chain Id of an auxiliary chain like `1406`, `1407` which can't be a name like `ropsten`. 
Refer [here](https://github.com/OpenST/mosaic-chains/blob/32708c3742c7ef7079e9347e14450ccb868b723c/src/NewChain/CliqueGenesis.ts#L13) 
> * `NodeOptions` (specifically, I am thinking of `parseOptions`)

Done
> 
> The framing with which I'm working is that we start with `chain` and that the `chainId` will eventually be set to the value of `chain` or some other value (e.g., 1 for "ethereum"). This is why I think it should be `newChain` in `mosaic-create` but it's fine to stay `newChainId` in `Initialization.initialize`.
> 
> I also wonder if it might be a bit simpler to manage this if there were some refactoring. In at least two places, maybe others, constructors/initialization functions use distinct `chainId` and`nodeDescription` parameters (e.g., [mosaic-create.ts#L29](https://github.com/sarvesh-ost/mosaic-chains/blob/rename_chainID_to_chain/src/bin/mosaic-create.ts#L29), [AuxiliaryChain.ts#L62](https://github.com/sarvesh-ost/mosaic-chains/blob/rename_chainID_to_chain/src/NewChain/AuxiliaryChain.ts#L62)). I'm not sure that should be done in this ticket, but I do think it might ease the burden of determining which instances should be changed. If not in this ticket, then there should probably be a ticket to refactor, regardless.
> 
> cc: @deepesh-kn

File: src/Node/GethNode.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (jasonklein) [reply src/Node/GethNode.ts]: > Yea, I agree readme should specify details about this.

@deepesh-kn: if you also agree, I will add a ticket ✏️

**Pull request merged**